sw/device/silicon_creator/lib/drivers/usb.h

```diff
@@ -0,0 +1,170 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_USB_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_USB_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/silicon_creator/lib/drivers/stdusb.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Flags for endpoint configuration
+ */
+typedef enum usb_endpoint_type {
+  /** Endpoint is an IN endpoint */
+  kUsbEndpointTypeIn = 1,
+  /** Endpoint is a OUT endpoint */
+  kUsbEndpointTypeOut = 2,
+  /** Endpoint can accept SETUPDATA (e.g. a control endpoint) */
+  kUsbEndpointTypeSetup = 4,
+  /** Endpoint is a CONTROL endpoint */
+  kUsbEndpointTypeControl = 7,
+} usb_endpoint_type_t;
+
+/**
+ * Flags for managing usb transfers
+ */
+typedef enum usb_transfer_flags {
+  /** Transfer direction is IN (device to host) */
+  kUsbTransferFlagsIn = 0,
+  /** Transfer direction is OUT (host to device) */
+  kUsbTransferFlagsOut = 1,
+  /** Transfer direction mask */
+  kUsbTransferFlagsDirMask = 1,
```

Comment:
Just a note here that typically, the direction is encoded in the MSB of the endpoint ID byte. But if you prefer this separate flag, I won't stand in your way, hehe.

In addition, IN and OUT do not describe _transfers_; they describe _transactions_.

Reply:
Fixed.